sympy/core/numbers.py

```diff
@@ -1368,26 +1359,43 @@ def __new__(cls, p, q=None, gcd=None):
             Q *= int(q)
         q = Q
 
+        if gcd is not None:
+            sympy_deprecation_warning(
+                "gcd is deprecated in Rational, use nsimplify instead",
+                deprecated_since_version="1.11",
+                active_deprecations_target="deprecated-rational-gcd",
+                stacklevel=4,
+            )
+            return cls._new(p, q, gcd)
+
         # p and q are now ints
+        return cls._new(p, q)
+
+    @classmethod
+    def _new(cls, p, q, gcd=None):
         if q == 0:
             if p == 0:
                 if _errdict["divide"]:
                     raise ValueError("Indeterminate 0/0")
                 else:
                     return S.NaN
             return S.ComplexInfinity
```

Comment:
This nan/zoo handling is also something that needs to be cleaned up for e.g. static typing. This should be handled in Mul rather than in the Rational constructor which should only ever return a valid Rational or raise an error.